README.md

```diff
@@ -4,221 +4,501 @@
+
+  - **Type:** String
+  - **Requirement:** Required when the [`user`](#client-param-user) parameter is set.
+    See [User Authentication](#user-authentication).
+
+- <a id="client-param-token"></a>**token** &mdash; a valid platform access key that allows access to the backend data.
+  To get this access key, select the user profile icon on any platform dashboard page, select **Access Tokens**, and copy an existing access key or create a new key.
+
+  - **Type:** String
+  - **Requirement:** Required when neither the [`user`](#client-param-user) or [`password`](#client-param-password) parameters or the authentication environment variables are set.
+    See [User Authentication](#user-authentication).
+
+<a id="client-constructor-example"></a>
+#### Example
+
+The following example, for local platform execution, creates a Frames client for accessing data in the "users" container by using the authentication credentials of the user "iguazio":
 
-All of frames operations are executed via the `client` object. To create a client object simply provide the Iguazio web-api endpoint and optional credentials.
 ```python
 import v3io_frames as v3f
-client = v3f.Client('framesd:8081', user='user1', password='pass')
+client = v3f.Client("framesd:8081", user="iguazio", password="mypass", container="users")
 ```
-Note: When running from within the managed jupyter notebook on the iguazio platform there is no need to add credentials as this is handled by the platform.  
-Next, for every operation we need to provide a `backend`, and a `table` parameters and optionally other function specific arguments.
 
-### Create
-Creates a new table for the desired backend. Not all backends require a table to be created prior to ingestion. For example KV table will be created while ingesting new data, on the other hand since TSDB tables have mandatory fields we need to create a table before ingesting new data.  
+<a id="client-common-method-params"></a>
+### Common Client Method Parameters
+
+All client methods receive the following common parameters:
+
+- <a id="client-method-param-backend"></a>**backend** &mdash; the backend data type for the operation.
+  See the backend-types descriptions in the [overview](#backend-types).
+
+  - **Type:** String
+  - **Valid Values:** `"csv"` | `"kv"` | `"stream"` | `"tsdb"`
+  - **Requirement:** Required
+
+- <a id="client-method-param-table"></a>**table** &mdash; the relative path to the backend data &mdash; a directory in the target platform data container (as configured for the client object) that represents a platform data collection, such as a TSDB or NoSQL table or a stream.
+  For example, `"mytable"` or `"examples/tsdb/my_metrics"`.
+
+  - **Type:** String
+  - **Requirement:** Required
+
+Additional method-specific parameters are described for each method.
+
+<a id="method-create"></a>
+### create Method
+
+Creates a new data collection (table/stream) in a platform data container according to the configured backend.
```

Comment:
According to the *given* backend.
Configured sounds static and complicated, in frames you specify the backend on every function call. 
(same for all other methods)

Reply:
I edited the method description, including to refer to "the specified backend type". The changes will be committed soon to the PR branch.